modules/imgproc/src/stackblur.cpp

```diff
@@ -0,0 +1,845 @@
+                    }
+
+                    stack[stack_start*cn] = *(src_new);
+                    stack[stack_start*cn + 1] = *(src_new + 1);
+                    stack[stack_start*cn + 2] = *(src_new + 2);
+
+                    sumIn[0] += static_cast<int>(*(src_new));
+                    sumIn[1] += static_cast<int>(*(src_new + 1));
+                    sumIn[2] += static_cast<int>(*(src_new + 2));
+
+                    sum[0] += sumIn[0];
+                    sum[1] += sumIn[1];
+                    sum[2] += sumIn[2];
+
+                    int sp1 = sp + 1;
+                    sp1 &= -(sp1 < div);
+
+                    temp = static_cast<int>(stack[sp1*cn]);
+                    sumOut[0] += temp;
+                    sumIn[0] -= temp;
+
+                    temp = static_cast<int>(stack[sp1*cn + 1]);
+                    sumOut[1] += temp;
+                    sumIn[1] -= temp;
+
+                    temp = static_cast<int>(stack[sp1*cn + 2]);
+                    sumOut[2] += temp;
+                    sumIn[2] -= temp;
+
+                    dst_ptr += cn;
+
+                    if (xp < wm)
+                    {
+                        xp++;
+                        src_ptr += cn;
+                    }
+
+                    ++sp;
+                    if (sp >= div) sp = 0;
+                }
+
+            }
+        }
+        else if (cn == 4)
+        {
+
+#if CV_SIMD
+            v_uint32x4 v_sum = v_setzero_u32();
+            v_uint32x4 v_sumIn = v_setzero_u32();
+            v_uint32x4 v_sumOut = v_setzero_u32();
+            v_uint32x4 v_temp = v_setzero_u32();
+            v_uint32x4 v_mul_sum = v_setall_u32((int) mul_sum);
+            v_uint32x4 v_stack[div];
+
+            for(int k = 0; k < div; k++)
+            {
+                v_stack[k] = v_setzero_u32();
+            }
+
+#endif
+            for (; row < range.end; row++)
+            {
+
+#ifdef CV_SIMD
+                v_sum = v_setzero_u32();
+                v_sumIn = v_setzero_u32();
+                v_sumOut = v_setzero_u32();
+#elif
+                memset(sum, 0, sizeof(unsigned int) * 4);
+                memset(sumIn, 0, sizeof(unsigned int) * 4);
+                memset(sumOut, 0, sizeof(unsigned int) * 4);
+                memset(stack, 0, sizeof(uchar) * 4 * div);
+#endif
+
+                src_ptr = src.data + src.step.p[0] * row;
+                for (i = 0; i <= radius; i++)
+                {
+#ifdef CV_SIMD
+                    v_stack[i] = v_load_expand_q(src_ptr);
+                    v_temp = v_stack[i];
+                    v_sum += v_temp * v_setall_u32(i + 1);
+                    v_sumOut += v_temp;
+#elif
+                    temp = static_cast<int>(*src_ptr);
+                    stack[i*cn] = *src_ptr;
+                    sum[0] += temp * (i + 1);
+                    sumOut[0] += temp;
+
+                    temp = static_cast<int>(*(src_ptr + 1));
+                    stack[i*cn + 1] = *(src_ptr + 1);
```

Comment:
replace cn with 4 in this branch, or define a constant "const int CN = 4;" and replace "cn" with "CN"